### PR TITLE
changed: Use toml file for versions of dependencies

### PR DESCRIPTION
**Pull request by pboos**
Use toml file for versioning as described in this gradle documentation: https://docs.gradle.org/current/userguide/platforms.html#sub:central-declaration-of-dependencies

This might probably help with automatic updates using dependabot. As it currently doesn't detect several updates like spring boot.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Testing
+mockito-core = { group = "org.mockito", name = "mockito-core", version.ref = "mockito" }
+mockito-junit-jupiter = { group = "org.mockito", name = "mockito-junit-jupiter", version.ref = "mockito" }
+junit-jupiter-api = { group = "org.junit.jupiter", name = "junit-jupiter-api", version.ref = "junit-jupiter" }
+junit-jupiter-engine = { group = "org.junit.jupiter", name = "junit-jupiter-engine", version.ref = "junit-jupiter" }
+
+
+[plugins]
+spring-boot = { id = "org.springframework.boot", version.ref = "spring-boot" }
+spring-dependencyManagement = { id = "io.spring.dependency-management", version.ref = "spring-dependency-management" }
```
Comment (catarina-correia) [inline gradle/libs.versions.toml:37]: Can we rename this to `spring-dependency-management` to be consistent?

File: build.gradle [hunk 2]
```diff
@@ -24,22 +24,22 @@ allprojects {
...
-            testImplementation "org.mockito:mockito-junit-jupiter:$mockitoVersion"
+            testImplementation(libs.junit.jupiter.api)
+            testImplementation(libs.junit.jupiter.engine)
+            testImplementation(libs.mockito.core)
+            testImplementation(libs.mockito.junit.jupiter)
         }
 
         jacoco {
-            toolVersion = jacocoVersion
+            toolVersion = "0.8.7"
```
Comment (catarina-correia) [inline build.gradle:42]: Can't we use something like [this](https://docs.gradle.org/current/userguide/platforms.html#sec:common-version-numbers)?
```
checkstyle {
    // will use the version declared in the catalog
    toolVersion = libs.versions.checkstyle.get()
}
```

**Review by catarina-correia — APPROVED** [review]
lgtm 🚀

File: build.gradle (hunk 2 above)
Comment (pboos) [reply build.gradle]: good point 👍 done ✅

**Pull request merged**